Mathlib/NumberTheory/LSeries/HurwitzZetaEven.lean

```diff
@@ -0,0 +1,391 @@
+/-
+Copyright (c) 2024 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import Mathlib.NumberTheory.LSeries.AbstractFuncEq
+import Mathlib.NumberTheory.ModularForms.JacobiTheta.Bounds
+
+/-!
+# Even Hurwitz zeta functions
+
+In this file we study the functions on `ℂ` which are the meromorphic continuation of the following
+series (convergent for `1 < re s`), where `a ∈ ℝ` is a parameter:
+
+`completedHurwitzZetaEven a s = 1 / 2 * □ * ∑' n : ℤ, 1 / |n + a| ^ s`
+
+and
+
+`completedCosZeta a s = □ * ∑' n : ℕ, cos (2 * π * a * n) / |n| ^ s`
+
+where `□` denotes a Gamma factor. Note that the term for `n = -a` in the first sum is omitted
+if `a` is an integer, and the term for `n = 0` is omitted in the second sum (always).
+
+Of course, we cannot *define* these functions by the above formulae (since existence of the
+meromorphic continuation is not at all obvious); we in fact construct them as Mellin transforms of
+various versions of the Jacobi theta function. We also define modified versions with a subscript `0`
+which are entire functions differing from the above by multiples of `1 / s` and `1 / (1 - s)`.
+
+## Main definitions and theorems
+
+* `completedHurwitzZetaEven`: the completed Hurwitz zeta function
+* `completedCosZeta`: the completed cosine zeta function
+* `differentiableAt_completedHurwitzZetaEven` and `differentiableAt_completedCosZeta`:
+  differentiability away from `s = 0` and `s = 1`
+* `completedHurwitzZetaEven_one_sub`: the functional equation
+  `completedHurwitzZetaEven a (1 - s) = completedCosZeta a s`
+* `completedHurwitzZetaEven_eq_tsum_int` and `completedCosZeta_eq_tsum_nat`: relation between the
+  zeta functions and corresponding Dirichlet series for `1 < re s`
+-/
+noncomputable section
+
+open Complex Filter Topology Asymptotics Real Set Classical MeasureTheory
+
+section kernel_defs
+/-!
+## Definitions and elementary properties of kernels
+-/
+
+/-- Even Hurwitz zeta kernel (function whose Mellin transform will be the even part of the
+completed Hurwit zeta function). See `evenKernel_def` for the defining formula, and
+`hasSum_int_evenKernel` for an expression as a sum over `ℤ`. -/
+@[irreducible] def evenKernel (a : UnitAddCircle) (x : ℝ) : ℝ :=
+  (show Function.Periodic
+    (fun ξ : ℝ ↦ rexp (-π * ξ ^ 2 * x) * re (jacobiTheta₂ (ξ * I * x) (I * x))) 1 by
+      intro ξ
+      simp only [ofReal_add, ofReal_one, add_mul, one_mul, jacobiTheta₂_add_left']
+      have : cexp (-↑π * I * ((I * ↑x) + 2 * (↑ξ * I * ↑x))) = rexp (π * (x + 2 * ξ * x)) := by
+        simp only [ofReal_exp, push_cast]
+        ring_nf
+        rw [I_sq]
+        ring_nf
```

Comment:
This is slightly shorter and quicker (although I can't figure out how to get the formatting of the preview to look right).

```suggestion
   have : cexp (-↑π * I * ((I * ↑x) + 2 * (↑ξ * I * ↑x))) = rexp (π * (x + 2 * ξ * x)) := by
        ring_nf
        simp only [I_sq, mul_neg, mul_one, neg_mul, neg_neg, sub_neg_eq_add, ofReal_exp, ofReal_add,
          ofReal_mul, ofReal_ofNat]
```